dask/array/routines.py

```diff
@@ -183,6 +183,23 @@ def diff(a, n=1, axis=-1):
     return r
 
 
+@wraps(np.ediff1d)
+def ediff1d(ary, to_end=None, to_begin=None):
+    ary = asarray(ary)
+
+    aryf = ary.flatten()
+    r = aryf[1:] - aryf[:-1]
+
+    r = [r]
+    if to_begin is not None:
+        r = [asarray(to_begin).flatten()] + r
+    if to_end is not None:
+        r = r + [asarray(to_end).flatten()]
+    r = concatenate(r)
```

Comment:
Can be optimized out if both `to_begin` and `to_end` are `None` using the proposed change to `concatenate` in PR ( https://github.com/dask/dask/pull/2610 ).